### PR TITLE
Drop DOT dependency

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Reimplements the GC debug output without the dot or gonum library. The format is simple enough to output without the libraries and it eliminates a ton of transitive dependencies.

```diff
{
-	"directDependencies": 216,
-	"transitiveDependencies": 255,
-	"totalDependencies": 308,
+	"directDependencies": 213,
+	"transitiveDependencies": 235,
+	"totalDependencies": 288,
 	"maxDepthOfDependencies": 30
 }
```

```release-note
NONE
```

xref https://github.com/kubernetes/kubernetes/pull/112688

/cc @deads2k @dims

### CODE REVIEW

**Comment by liggitt** [comment]
```
kubectl create deployment --image test test1
kubectl create deployment --image test test2
kubectl create deployment --image test test3
kubectl scale deployment/test2 --replicas 2
kubectl scale deployment/test3 --replicas 3
kubectl get --insecure-skip-tls-verify --server https://localhost:10257 --raw /debug/controllers/garbagecollector/graph | dot -ograph.pdf -Tpdf
```

Result: [graph.pdf](https://github.com/kubernetes/kubernetes/files/9630505/graph.pdf)

**Comment by chenfanlinux** [comment]
test

**Comment by pacoxu** [comment]
> test

@chenfanlinux  I would suggest that you should play in https://github.com/kubernetes-sigs/contributor-playground project.

File: pkg/controller/garbagecollector/dump.go [hunk 1]
```diff
@@ -106,106 +120,120 @@ namespace=%v
...
+		{Key: "beingDeleted", Value: fmt.Sprintf(`%v`, v.beingDeleted)},
+		{Key: "deletingDependents", Value: fmt.Sprintf(`%v`, v.deletingDependents)},
+		{Key: "virtual", Value: fmt.Sprintf(`%v`, v.virtual)},
 	}
 }
 
-// NewGonumVertex creates a new gonumVertex.
-func NewGonumVertex(node *node, nodeID int64) *gonumVertex {
+// NewdotVertex creates a new dotVertex.
+func NewdotVertex(node *node) *dotVertex {
```
Comment (deads2k) [inline pkg/controller/garbagecollector/dump.go:140]: `NewDOTVertex` to match other abbreviations.

**Comment by deads2k** [comment]
not sure why the tests are commented out, but rendering in-line instead of with the library is ok. We can use external graph tools to query more deeply.

**Comment by liggitt** [comment]
> not sure why the tests are commented out, but rendering in-line instead of with the library is ok. We can use external graph tools to query more deeply.

was quick and dirty and wanted to make sure the direction was acceptable before I fixed up the tests

File: pkg/controller/garbagecollector/dump.go (hunk 1 above)
Comment (liggitt) [reply pkg/controller/garbagecollector/dump.go]: done

**Comment by liggitt** [comment]
fixed up tests, added marshaling tests as well

**Comment by dims** [comment]
unrelated failure, running again
```
{Failed  === RUN   TestConnectionPings
    connection_test.go:240: server: failed to send any pings (check logs)
--- FAIL: TestConnectionPings (0.08s)
}
```

/test pull-kubernetes-unit

**Comment by dims** [comment]
Numbers dropped as expected!

```
--- /logs/artifacts/stats-base.json	2022-09-23 17:16:59.493835632 +0000
+++ /logs/artifacts/stats.json	2022-09-23 17:16:31.155207769 +0000
@@ -1,6 +1,6 @@
 {
-	"directDependencies": 216,
-	"transitiveDependencies": 255,
-	"totalDependencies": 308,
+	"directDependencies": 213,
+	"transitiveDependencies": 235,
+	"totalDependencies": 288,
 	"maxDepthOfDependencies": 30
 }
```

**Comment by dims** [comment]
image

/approve
/lgtm

thanks Jordan!

**Pull request merged**